packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+   *
+   * Exactly one of `launchTemplateId` and `launchTemplateName` may be set.
+   *
+   * @default None
+   */
+  readonly launchTemplateId?: string;
+
+  /**
+   * The name of the Launch Template
+   *
+   * Exactly one of `launchTemplateId` and `launchTemplateName` may be set.
+   *
+   * @default None
+   */
+  readonly launchTemplateName?: string;
+}
+
+/**
+ * This represents an EC2 LaunchTemplate.
+ *
+ * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-launch-templates.html
+ */
+export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGrantable, IConnectable {
+  /**
+   * Import an existing LaunchTemplate.
+   */
+  public static fromLaunchTemplateAttributes(scope: Construct, id: string, attrs: LaunchTemplateAttributes): ILaunchTemplate {
+    const haveId = Boolean(attrs.launchTemplateId);
+    const haveName = Boolean(attrs.launchTemplateName);
+    if (haveId == haveName) {
+      throw new Error('LaunchTemplate.fromLaunchTemplateAttributes() requires exactly one of launchTemplateId or launchTemplateName be provided.');
+    }
+
+    class Import extends Resource implements ILaunchTemplate {
+      public readonly versionNumber = attrs.versionNumber ?? LaunchTemplateSpecialVersions.DEFAULT_VERSION;
+      public readonly launchTemplateId? = attrs.launchTemplateId;
+      public readonly launchTemplateName? = attrs.launchTemplateName;
+    }
+    return new Import(scope, id);
+  }
+
+  // ============================================
+  //   Members for ILaunchTemplate interface
+  /**
+   * @inheritdoc
+   */
+  public readonly versionNumber: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateId?: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateName?: string;
+
+  // =============================================
+  //   Data members
+
+  /**
+   * The default version for the launch template.
+   *
+   * @attribute
+   */
+  public readonly defaultVersionNumber: string;
+
+  /**
+   * The latest version of the launch template.
+   *
+   * @attribute
+   */
+  public readonly latestVersionNumber: string;
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Principal to grant permissions to.
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * Note: If the LaunchTemplate was not created with a SecurityGroup, then this connections
+   * object will be empty and operations on it will do nothing.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * UserData executed by instances that are launched from this template.
+   */
+  public readonly userData?: UserData;
+
+
+  protected tagSpecifications: LaunchTemplateTagSpecification[];
+
+  // =============================================
+
+  constructor(scope: Construct, id: string, props?: LaunchTemplateProps) {
```

Comment:
If you declare this as `props: LaunchTemplateProps = {}` it saves a bunch of `props?.` operators in this method.